datumaro/components/hl_ops/__init__.py

```diff
@@ -0,0 +1,231 @@
+
+        if isinstance(method, str):
+            if env is None:
+                env = Environment()
+            method = env.transforms[method]
+
+        if not (inspect.isclass(method) and issubclass(method, Transform)):
+            raise TypeError("Unexpected 'method' argument type: %s" % type(method))
+
+        produced = method(dataset, **kwargs)
+
+        return Dataset(source=produced, env=env)
+
+    @staticmethod
+    def filter(
+        dataset: IDataset,
+        expr: str,
+        *,  # pylint: disable=redefined-builtin
+        filter_annotations: bool = False,
+        remove_empty: bool = False,
+    ) -> IDataset:
+        """
+        Filters out some dataset items or annotations, using a custom filter
+        expression.
+
+        Args:
+            dataset: The dataset to be filtered
+            expr: XPath-formatted filter expression
+                (e.g. `/item[subset = 'train']`, `/item/annotation[label = 'cat']`)
+            filter_annotations: Indicates if the filter should be
+                applied to items or annotations
+            remove_empty: When filtering annotations, allows to
+                exclude empty items from the resulting dataset
+
+        Returns: a wrapper around the input dataset, which is computed lazily
+            during iteration
+        """
+
+        if filter_annotations:
+            return HLOps.transform(
+                dataset, XPathAnnotationsFilter, xpath=expr, remove_empty=remove_empty
+            )
+        else:
+            if not expr:
+                return dataset
+            return HLOps.transform(dataset, XPathDatasetFilter, xpath=expr)
+
+    @staticmethod
+    def merge(*datasets: IDataset) -> IDataset:
+        """
+        Merges several datasets using the "simple" (exact matching) algorithm:
+
+            - items are matched by (id, subset) pairs
+            - matching items share the fields available
+
+                - nothing + nothing = nothing,
+                - nothing + something = something
+                - something A + something B = conflict
+            - annotations are matched by value and shared
+            - in case of conflicts, throws an error
+
+        Returns: a wrapper around the input datasets
+        """
+
+        merger = ExactMerge()
+        infos = merger.merge_infos(d.infos() for d in datasets)
+        categories = merger.merge_categories(d.categories() for d in datasets)
+        media_type = merger.merge_media_types(datasets)
+        return DatasetItemStorageDatasetView(
+            merger.merge(*datasets), infos=infos, categories=categories, media_type=media_type
+        )
+
+    @staticmethod
+    def run_model(
+        dataset: IDataset,
+        model: Union[Launcher, Type[ModelTransform]],
+        *,
+        batch_size: int = 1,
+        append_annotation: bool = False,
+        **kwargs,
+    ) -> IDataset:
+        """
+        Applies a model to dataset items' media and produces a dataset with
```

Comment:
```suggestion
        Applies a model to dataset item's media and produces a dataset with
```

Reply:
Revised the model docstring for more clearance as follows.

Run the model on the dataset item media entities, such as images, to obtain pseudo labels and add them as dataset annotations.